optaplanner-core/src/main/java/org/optaplanner/core/api/score/holder/AbstractScoreHolder.java

```diff
@@ -104,6 +107,14 @@ public void configureConstraintWeight(Rule rule, Score_ constraintWeight) {
         }
     }
 
+    /**
+     * Requires @{@link OptaPlannerRuleEventListener} to be added as event listener on {@link KieSession}, otherwise the
+     * score changes caused by the constraint matches would not be undone. See
+     * {@link DroolsScoreDirector#resetKieSession()} for an example.
+     * @param kcontext The rule for which to register the match.
```

Comment:
nitpick (in according to sun's javadoc styleguides):
"The rule for which to register the match." is not a sentence - it doesn't have a verb in the main part - so it shouldn't  end with a dot. It's ok for javadocs parameters and return types not to be sentances - but a senstence-ending character (. or ? or !) can only used if it's a sentence.
=> @param kcontext The rule for which to register the match